tests/test_numpy_dtypes.cpp

```diff
@@ -70,6 +70,13 @@ struct StringStruct {
     std::array<char, 3> b;
 };
 
+struct ArrayStruct {
+    char a[3][4];
```

Comment:
By the way, that wouldn't prevent using a struct with zero-sized arrays in it - you just couldn't reflect the zero-sized array into Python. That's no great loss given that the field contains no information anyway.